### PR TITLE
Deprecate proto definition on gamm & twap v1beta1

### PR DESCRIPTION
**Pull request by iboss-ptk**
## What is the purpose of the change

Mark proto related to `gamm` `SpotPrice` and `twap` `ArithmeticTwap` & `ArithmeticTwapToNow` as deprecated so that client code gen has information to mark deprecated on generated code.

### CODE REVIEW

File: x/gamm/client/cli/query.go [hunk 1]
```diff
@@ -388,6 +388,7 @@ func GetCmdSpotPrice() *cobra.Command {
 				return err
 			}
 
+			// nolint: staticcheck
```
Comment (mattverse) [inline x/gamm/client/cli/query.go:391]: What's the reason for all the nolint static checks that has been added?

**Review by mattverse — APPROVED** [review]
Agreed with adding the deprecation fields!

File: wasmbinding/stargate_whitelist.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ func init() {
 	setWhitelistedQuery("/osmosis.gamm.v1beta1.Query/PoolParams", &gammtypes.QueryPoolParamsResponse{})
 	setWhitelistedQuery("/osmosis.gamm.v1beta1.Query/TotalPoolLiquidity", &gammtypes.QueryTotalPoolLiquidityResponse{})
 	setWhitelistedQuery("/osmosis.gamm.v1beta1.Query/TotalShares", &gammtypes.QueryTotalSharesResponse{})
-	setWhitelistedQuery("/osmosis.gamm.v1beta1.Query/SpotPrice", &gammtypes.QuerySpotPriceResponse{})
+	setWhitelistedQuery("/osmosis.gamm.v1beta1.Query/SpotPrice", &gammtypes.QuerySpotPriceResponse{}) // nolint: staticcheck
```
Comment (ValarDragon) [inline wasmbinding/stargate_whitelist.go:82]: Oh what, did someone accidentally add this back? This should've been deleted here

File: x/gamm/client/cli/query.go (hunk 1 above)
Comment (iboss-ptk) [reply x/gamm/client/cli/query.go:391]: it's warns on the use of deprecated structs. I think some might be true positive, can you help having a second look on which ones are?

**Pull request merged**